core/src/main/java9/feign/DefaultMethodHandler.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2012-2021 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign;
+
+import feign.InvocationHandlerFactory.MethodHandler;
+import org.jvnet.animal_sniffer.IgnoreJRERequirement;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.reflect.Method;
+
+/**
+ * Handles default methods by directly invoking the default method code on the interface. The bindTo
+ * method must be called on the result before invoke is called.
+ */
+@IgnoreJRERequirement
+final class DefaultMethodHandler implements MethodHandler {
+  // Uses Java 9 MethodHandle based reflection. As default methods will only exist when
+  // run on a Java 8 JVM this will not affect use on legacy JVMs.
+  // When Feign upgrades to Java 7, remove the @IgnoreJRERequirement annotation.
+  private final MethodHandle unboundHandle;
+
+  // handle is effectively final after bindTo has been called.
+  private MethodHandle handle;
+
+  public DefaultMethodHandler(Method defaultMethod) {
+    try {
+      final Class<?> declaringClass = defaultMethod.getDeclaringClass();
+      final Lookup lookup = MethodHandles.privateLookupIn(declaringClass, MethodHandles.lookup());
```

Comment:
I think I have a solution for this...

can we just use reflection on the existing `DefaultMethodHandler`?

We just need try to invoke the new API using reflection, if that fail, fallback to existing code

Reply:
That’s what I’ve bee saying

Reply:
Will create a pr for this tonight

Reply:
I just did, had some time allocated at work to be around feign =)

Reply:
The changes look good! Really like the use of `deny` to get java 11 to help test this change.